nixos/modules/system/boot/networkd.nix

```diff
@@ -712,6 +712,9 @@ in
     systemd.services.systemd-networkd = {
       wantedBy = [ "multi-user.target" ];
       restartTriggers = map (f: f.source) (unitFiles);
+      # prevent race condition with interface renaming (#39069)
+      requires = [ "systemd-udev-settle.service" ];
+      after = [ "systemd-udev-settle.service" ];
```

Comment:
I agree this is not ideal but the best we can do in stage 2 to prevent the race condition. We can revert this as soon as we do the renaming in stage 1.

Reply:
Instead of using `systemd-udev-settle`, we could add 
```serviceConfig.ExecStartPre="${systemd.package}/bin/udevadm settle -t 10"; ```
directly to `systemd-networkd` to use a shorter timeout than the default 120s. I guess 10s will be enough to get interface renaming done...